src/vorta/views/partials/password_input.py

```diff
@@ -0,0 +1,127 @@
+
+    def __init__(self, parent=None, show_visibility_button=True):
+        super().__init__(parent)
+
+        self._show_visibility_button = show_visibility_button
+
+        self.setEchoMode(QLineEdit.Password)
+
+        if self._show_visibility_button:
+            self.showHideAction = QAction(self.tr("Show password"), self)
+            self.showHideAction.setCheckable(True)
+            self.showHideAction.toggled.connect(self._toggle_password_visibility)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+            self.addAction(self.showHideAction, QLineEdit.TrailingPosition)
+
+    def get_password(self):
+        """Return password text"""
+        return self.text()
+
+    def _toggle_password_visibility(self):
+        if self.echoMode() == QLineEdit.Password:
+            self.setEchoMode(QLineEdit.Normal)
+            self.showHideAction.setIcon(get_colored_icon("eye"))
+        else:
+            self.setEchoMode(QLineEdit.Password)
+            self.showHideAction.setIcon(get_colored_icon("eye-slash"))
+
+    def set_error_state(self, error):
+        """Display red border if password is not valid"""
+        if error:
+            self.setStyleSheet('border: 2px solid red;')
+        else:
+            self.setStyleSheet('')
+
+
+class PasswordInput(QWidget):
+    # password label changed signal
+    passwordLabelChanged = QtCore.pyqtSignal(str)
+
+    def __init__(self, parent=None, form_layout=None, minimum_length=9, show_error=True, label: list = None):
+        super().__init__(parent)
+        self._minimum_length = minimum_length
+        self._show_error = show_error
+
+        if label:
+            self._label_1 = QLabel(label[0])
+            self._label_2 = QLabel(label[1])
+        else:
+            self._label_1 = QLabel(self.tr("Enter passphrase:"))
+            self._label_2 = QLabel(self.tr("Confirm passphrase:"))
+
+        # Create password line edits
+        self.passwordLineEdit = PasswordLineEdit(parent=self)
+        self.confirmLineEdit = PasswordLineEdit(parent=self)
+        self.password_label = QLabel("")
+
+        self.passwordLineEdit.editingFinished.connect(self.validate)
+        self.confirmLineEdit.textChanged.connect(self.validate)
+
+        # form_layout = parent.layout() if isinstance(parent, QFormLayout) else None
+        if form_layout is not None:
+            form_layout.addRow(self._label_1, self.passwordLineEdit)
+            form_layout.addRow(self._label_2, self.confirmLineEdit)
+            form_layout.addRow(self.password_label)
+        else:
+            password_form = QFormLayout(self)
+            password_form.setContentsMargins(0, 0, 0, 0)
+            password_form.addRow(self._label_1, self.passwordLineEdit)
+            password_form.addRow(self._label_2, self.confirmLineEdit)
+            password_form.addRow(self.password_label)
+
+            password_form.setFieldGrowthPolicy(QFormLayout.AllNonFixedFieldsGrow)
```

Comment:
Let's move this into a method, so that we can instantiate the widget and add it to a form later.

Reply:
:+!: